### ISSUE TITLE
instances of MonoLogger passed to call of mdadm

### ISSUE THREAD

**Issue by gretel** [atrull/ephemeral_raid_cookbook#3]
cookbook `0.0.9` on `chef-12.5.1` on `ubuntu-12.04` on `EC2 c3.4xlarge`: seems like something goes really wrong

```
           Error executing action `create` on resource 'mdadm[/dev/md0]'
           ================================================================================

           Mixlib::ShellOut::ShellCommandFailed
           ------------------------------------
           Expected process to exit with [0], but received '1'
           ---- Begin output of yes | mdadm --create /dev/md0 --level 0 --chunk=256 --metadata=0.90 --raid-devices 4 /dev/xvdb /dev/xvdc #<MonoLogger:0x000000046e5358> #<MonoLogger:0x000000046e5358> ----
       STDOUT:
           STDERR: mdadm: You haven't given enough devices (real or missing) to create this array
           ---- End output of yes | mdadm --create /dev/md0 --level 0 --chunk=256 --metadata=0.90 --raid-devices 4 /dev/xvdb /dev/xvdc #<MonoLogger:0x000000046e5358> #<MonoLogger:0x000000046e5358> ----
       Ran yes | mdadm --create /dev/md0 --level 0 --chunk=256 --metadata=0.90 --raid-devices 4 /dev/xvdb /dev/xvdc #<MonoLogger:0x000000046e5358> #<MonoLogger:0x000000046e5358> returned 1

           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/ephemeral_raid/recipes/makeraid.rb

            36:   mdadm "#{node[:ephemeral][:raid][:device]}" do
            37:     devices ephemeral_devices
            38:     chunk node[:ephemeral][:raid][:chunk_size] if node[:ephemeral][:raid][:chunk_size]
        39:     level 0
            40:     action :create
        41:     notifies :run, "execute[set_readahead_on_device_after_creation]", :immediately
        42:     only_if "which mdadm"
            43:   end
            44: else

           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/ephemeral_raid/recipes/makeraid.rb:36:in `from_file'

           mdadm("/dev/md0") do
             action [:create]
             retries 0
         retry_delay 2
             default_guard_interpreter :default
             chunk 256
         devices ["/dev/xvdb", "/dev/xvdc", [#<MonoLogger:0x000000046e5358 @progname=nil, @level=2, @default_formatter=#<Logger::Formatter:0x000000046e5560 @datetime_format=nil>, @formatter=#<Mixlib::Log::Formatter:0x000000046e7b30 @datetime_format=nil>, @logdev=#<MonoLogger::LocklessLogDevice:0x000000046e5628 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>>>], [#<MonoLogger:0x000000046e5358 @progname=nil, @level=2, @default_formatter=#<Logger::Formatter:0x000000046e5560 @datetime_format=nil>, @formatter=#<Mixlib::Log::Formatter:0x000000046e7b30 @datetime_format=nil>, @logdev=#<MonoLogger::LocklessLogDevice:0x000000046e5628 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>>>]]
         level 0
             metadata "0.90"
             raid_device "/dev/md0"
         declared_type :mdadm
             cookbook_name "ephemeral_raid"
             recipe_name "makeraid"
         only_if "which mdadm"
           end
```

**Comment by atrull** [comment]
Hi Tom,

I notice from the output it is trying to raid 4 raid devices ? But it is only finding two. I'm guessing this may be related to the block device mapping question you asked separately.

Can you let me know how many devices you mapped - was it just the ephemerals or were there more (ebs volumes rather than instance store) ? And also "ls /dev/xvd*" ?

Thanks

Alex

**Comment by atrull** [comment]
sorry to clarify, it is trying to make a raid 0 out of 4 devices but only two are managing to get listed.

**Comment by gretel** [comment]
@atrull yep, it's should be a `RAID0` containing of two devices, no spares.
it's a `c3.4xlarge` - has two SSDs. mapping wasn't change afaik, but gonna ask.
so the two ones declared are the only ones - besides a root device `sda`. i guess but i am not sure that there is one `EBS` device mapped - that might give 4 in total?

**Comment by atrull** [comment]
If you didn't modify the mapping, it could be that knife already did it for you (smartness) or.. that amazon saw the light and started mapping these things by default. Doesn't matter too much. What matters if understanding where the number 4 is coming from:

Alright. This will take a little curling to get info on...

curl http://169.254.169.254/latest/meta-data/block-device-mapping

I'm kicking off a c3.4xlarge to check, in eu-central-1. Let me know what you get from the above command..

**Comment by gretel** [comment]
@atrull great thanks; i got home already and don't have access right now..

**Comment by atrull** [comment]
With the two ephemeral ssds mapped I got two ephemeral devices back..

btw note it needs the / on the end.

curl http://169.254.169.254/latest/meta-data/block-device-mapping/
ami
ephemeral0
ephemeral1

**Comment by atrull** [comment]
Interestingly... without the mapping it does map them, but to sdb and sdc. This is new behaviour. It should not affect the cookbook however. So lets see what you get on your side when you check with curl.

I'll shut the other issue for now, but leave this open until you can access :-)

**Comment by gretel** [comment]
@atrull alright, thanks!

**Comment by atrull** [comment]
Hi Tom - just a reminder that this issue is open :-)

**Comment by gretel** [comment]
i'm on holidays.. be back at work next week.. regards

**Comment by atrull** [comment]
This should be resolved on 0.0.11.
